### DISCUSSION TITLE
[General] <eth_getBalance> {"jsonrpc":"2.0","error":{"code":-32602,"message":"Block with such an ID is pruned; the first retained block is 43847192"},"id":1}

### DISCUSSION THREAD

**Discussion by jun-layer0** [zkSync-Community-Hub/zksync-developers#739] [General] [answered]
### Environment

Mainnet

### L2 block number

0x1B4

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

`eth_getBalance` fails with the following error.
```
curl -H 'content-type:application/json' -d '{"id": 1, "jsonrpc": "2.0", "method": "eth_getBalance", "params": ["0x742d35Cc6634C0532925a3b844Bc454e4438f44e", "0x1B4"]}' http://localhost

{"jsonrpc":"2.0","error":{"code":-32602,"message":"Block with such an ID is pruned; the first retained block is 43847192"},"id":1}
```
We would like to run our mainnet node as an archival node.
Please let us know how we can make our node as an archive node.
Thank you

**Comment by alhenaai** [comment]
Hello!

To resolve the issue you're encountering with the `eth_getBalance` method and to configure your node as an archival node, you need to adjust the pruning behavior of your node. This involves changing specific configurations to ensure that your node retains all historical blocks instead of pruning them.

Here’s how you can adjust the configurations to change the pruning behavior:

1. **Access your node's configuration settings**: This might involve modifying configuration files or settings within your node's software.
2. **Adjust the pruning settings**: Set the node to retain all historical blocks. The exact setting will depend on the node software you are using.

Unfortunately, the specific steps to make these adjustments depend on the node software you are using, which isn't specified in your query. You might need to consult the documentation for your specific node software or reach out to their support for detailed instructions on setting up an archival node.

For more information on node setup and configurations, you might find this [discussion on setting up an archive node](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/735) helpful.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by vasyl-ivanchuk** [comment, marked as answer]
hey @jun-layer0 👋 

It seems that you have either used a snapshot or have pruning enabled. However, as mentioned in our [docs](https://github.com/matter-labs/zksync-era/blob/main/docs/guides/external-node/08_pruning.md#interaction-with-snapshot-recovery):
> If you need a node with the entire rollup history, using a Postgres dump is the only option, and pruning should be disabled.

I'd recommend going through the External Node [documentation](https://github.com/matter-labs/zksync-era/blob/main/docs/guides/external-node/00_quick_start.md). It covers everything, including how to initialize an EN as an archival node from a Postgres dump, and explains what pruning is and how to configure it.

**Reply by jun1suzuki** [reply]
Thank you for sharing the info.
We have `EN_PRUNING_ENABLED` flag set to `false`.
We will review the provided doc and go through the Postgres dump option.
Thank you

**Reply by lz-reery** [reply]
hey @vasyl-ivanchuk thank you for the answer! also working with Jun on this.
We just restored from https://storage.googleapis.com/zksync-data-restore/data-restore-mainnet.dump 
but the external node complains that `relation "blocks" already exists`
Two questions:
- Can you confirm that this is the correct postgres dump?
- Is there a way to configure the EN to only run the new migrations?
Thanks in advance! Also if you have more detailed doc on how to start from a pgdump it would be great
Cheers
